wlauto/workloads/adobereader/uiauto/app/src/main/java/com/arm/wlauto/uiauto/adobereader/UiAutomation.java

```diff
@@ -102,6 +102,10 @@ private void dismissWelcomeView() throws Exception {
         if (!onboarding_finish_button.exists()) {
             welcomeView.swipeLeft(10);
         }
+        
+        if (!onboarding_finish_button.exists()) {
+        return;
+        }
```

Comment:
That is true. My thinking behind the alternative way was that the OpenFile function starts with the assumption that the workload is on the correct screen and therefore the purpose of the dismissWelcomeView function should be to ensure that the workload is in the known state before continuing?
And also, potentially in the future there may be additional popups which may require new checks and the early return could be misleading.